### ISSUE TITLE
ProgressIndicator: ARIA progressbar nodes must have an accessible name (aria-progressbar-name)

### ISSUE THREAD

**Issue by daramful** [microsoft/fluentui#16251]
### Environment Information

- **Package version(s)**: 
  - npm version: 6.14.8
  - node version: v14.15.1
  - "office-ui-fabric-react": "7.153.2",
  - "jest-axe": "^4.1.0",

- **Browser and OS versions**: 
  - Microsoft Edge 87.0.664.57 (Official build) (64-bit)
  - Windows 10 Enterprise - 20H2 v19042.685

### Describe the issue:

While using axe-core to go through accessibility check, I run into an error message that says "ARIA progressbar nodes must have an accessible name (aria-progressbar-name)". Upon looking at the [rule info](https://dequeuniversity.com/rules/axe/4.1/aria-progressbar-name), I found there are primarily three solutions for this, which cannot be customized with current codebase. I know ProgressIndicator is eventually going to be replaced with ProgressBar, but I do want to see this fixed for accessibility enhancement.

### Please provide a reproduction of the issue in a codepen:

source code: https://github.com/microsoft/fluentui/blob/master/packages/react-internal/src/components/ProgressIndicator/ProgressIndicator.base.tsx#L95

#### Actual behavior:
```
  ● ProgressIndicator Test 

    expect(received).toHaveNoViolations(expected)

    Expected the HTML found at $('.progressBar') to have no violations:

    <div class="progressBar" role="progressbar"></div>

    Received:

    "ARIA progressbar nodes must have an accessible name (aria-progressbar-name)"

    Fix any of the following:

-       aria-label attribute does not exist or is empty
-       aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
-       Element has no title attribute
```

#### Expected behavior:

Accessibility resolved. 

### Documentation describing expected behavior

https://dequeuniversity.com/rules/axe/4.1/aria-progressbar-name

**Comment by ecraig12345** [comment]
Thanks for filing this and for the PR!

It seems odd that the [Accessibility Insights](https://accessibilityinsights.io/) browser extension we use as part of a11y passes doesn't flag this, since I think it uses the same underlying library. I checked the [`progressbar` ARIA spec](https://www.w3.org/TR/wai-aria-1.1/#progressbar) and it does state that an accessible name is required for the `progressbar` role (so it seems like flagging this is correct). The [full name calculation spec](https://www.w3.org/TR/accname-1.1/#mapping_additional_nd_te) includes several additional options for defining the name beyond what the axe-core rule suggests, but as far as I can tell we're not implementing any of them right now.

Rather than adding `title={ariaValueText}` to the `progressbar` element, I think it would be better to make the existing `label` prop be properly associated as the `progressbar`'s label (unless @smhigley has a better idea). You can do this by giving the label element an ID and passing that ID as the `progressbar`'s `aria-labelledby`. :
```tsx
import { classNamesFunction, getId } from '../../Utilities';
// ...
export class ProgressIndicatorBase extends React.Component<IProgressIndicatorProps, {}> {
  // ...
  private _labelId = getId(); // this generates a unique ID

  public render() {
    // ...
          <div className={classNames.itemName} id={this._labelId}>
            {label}
          </div>
    // ...
  }

  private _onRenderProgress = (props: IProgressIndicatorProps): JSX.Element => {
    // ...
        <div
          className={classNames.progressBar}
          style={progressBarStyles}
          role="progressbar"
          aria-labelledby={this.props.label ? this._labelId : undefined}
    // ...
```

**Comment by daramful** [comment]
This is actually resolved with ariaValueText attribute. I will close this.
